src/main/java/com/lorepo/icplayer/client/printable/PrintableController.java

```diff
@@ -50,7 +51,34 @@ private native JavaScriptObject initJSObject(PrintableController x) /*-{
 		controller.nextInt = function(upperBound) {
 			return x.@com.lorepo.icplayer.client.printable.PrintableController::nextInt(I)(upperBound);
 		};
+
+		controller.getScore = function() {
+			return x.@com.lorepo.icplayer.client.printable.PrintableController::scoreJS;
+		};
 		
 		return controller;
 	}-*/;
+
+	public String getPageId () {
+		return page.getId();
+	}
+
+	public void setScore(String score) {
+		this.scoreJS = parseJson(score);
+	}
+
+	private native JavaScriptObject parseJson(String raw)/*-{
+		try {
+		var score = JSON.parse(raw);

```

Comment:
brakuje wciec

Reply:
A to właściwie nie mój kod, więc zostawię tak jak jest, żeby nie było później konfliktów. To jest kod zaczerpnięty z brancha, od którego mieliśmy się odbić: https://github.com/icplayer/icplayer/pull/503/